File: engine/engine-model/src/main/scala/tools/packet/foreigneffect/ShowItemEffect.scala
package tools.packet.foreigneffect

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowItemEffect(private var _characterId: Int, private var _itemId: Int) extends PacketInput {
  def characterId: Int = _characterId

  def itemId: Int = _itemId

  override def opcode(): SendOpcode = SendOpcode.SHOW_ITEM_EFFECT
}

File: engine/engine-model/src/main/java/tools/packet/foreigneffect/ShowItemEffect.java
package tools.packet.foreigneffect;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowItemEffect(Integer characterId, Integer itemId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_ITEM_EFFECT;
   }
}